boards/shields/wnc_m14a2a/boards/frdm_k64f.conf

```diff
@@ -0,0 +1 @@
+CONFIG_UART_MCUX_2=y
```

Comment:
What if instead we add a CONFIG_ARDUINO_SERIAL to `drivers/serial/Kconfig`, and then the boards can intelligently enable the right instance like they do with the console uart? Something like this in `boards/arm/frdm_k64f/Kconfig.defconfig`

```
if UART_MCUX

config UART_MCUX_0
        default y if UART_CONSOLE
config UART_MCUX_2
        default y if ARDUINO_SERIAL
config UART_MCUX_3
        default y if BT_UART

endif # UART_MCUX
```

Reply:
@MaureenHelm I like this idea.  I was struggling with the abstraction between a shield's UART (arduino_serial) and a HW specific UART (Like if the driver needed to be used directly).

A great example of this is the Particle.io Boron board which has a U-blox SARA-R41xx modem connected to the nRF52840 directly via UART1 vs. a SparkFun LTE Cat M1 shield which has the U-blox SARA-R41xx connected to Arduino R3-compatible headers.

The shield configuration should only reference the arduino_serial and stay away from anything too specific to the device.

However, this approach won't work for the WNC-M14A2A modem.   The UART pins are on non-standard Arduino pins.  The traditional Arduino UART is actually C16/C17 pins (UART_3) for K64F.

Technically, ARDUINO_SERIAL should enable UART_3 (where BT_UART is identified).   And, the BT_UART setting could go away if we migrate that to a shield setup.